### PR TITLE
Rework Dockerfile

### PR DESCRIPTION
**Pull request by jsf9k**
# 

## 🗣 Description

Clean up the `Dockerfile` a la the changes recently made in cisagov/scanner#34 and cisagov/scanner#36.

This involves getting rid of the deprecated `MAINTAINER` tag and moving to a multistage build.

## 💭 Motivation and Context

These changes update the Docker image to use Debian Buster, remove the deprecated `MAINTAINER` tag, and leverage a multistage build to better organize the `Dockerfile` contents.

## 🧪 Testing

Verified that the new image and the previous one both generate the same list of hosts to be gathered.

## 🚥 Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by felddy — APPROVED** [review]
Nice!  Multi-stage FTW.   I'm glad you've done this, so I can crib.  
🦃 👍

File: Dockerfile [hunk 1]
```diff
@@ -1,52 +1,95 @@
...
-RUN pip install --upgrade \
+###
+RUN pip install --no-cache-dir --upgrade \
     docopt \
     https://github.com/cisagov/mongo-db-from-config/tarball/develop
 
+###
+# Remove build dependencies
+###
+RUN apt-get remove --quiet --quiet $BUILD_DEPS
```
Comment (mcdonnnj) [inline Dockerfile]: Was this supposed to be INSTALL_DEPS?

**Review by mcdonnnj — CHANGES_REQUESTED** [review]
Nice rework, I'm digging the multi-stage 💪 Please see my comment about that variable though.

File: Dockerfile (hunk 1 above)
Comment (jsf9k) [reply Dockerfile]: Good catch!  Fixed in 1271197.

**Review by mcdonnnj — APPROVED** [review]
Nice job 🤙

File: Dockerfile [hunk 2]
```diff
@@ -1,52 +1,95 @@
...
-    && adduser --system --gecos "Gatherer user" --group gatherer \
-    && chown -R gatherer:gatherer ${GATHERER_HOME}
-
-##
+###
 # Make sure pip and setuptools are the latest versions
-##
-RUN pip install --upgrade pip setuptools
+###
+RUN pip install --no-cache-dir --upgrade pip setuptools
```
Comment (dav3r) [inline Dockerfile:40]: May want to add a comment somewhere in here about why we aren't using a pip cache (similar to what you did over [here](https://github.com/cisagov/saver/pull/46/commits/6781a3e8af53a3bbb8157827717144e8f0882d89)).

**Review by dav3r — APPROVED** [review]
LGTM, with one comment that will sound very familiar
🍪

File: Dockerfile (hunk 2 above)
Comment (jsf9k) [reply Dockerfile:40]: Remedied in commits 546f23f, cisagov/scanner@2b265d6, and cisagov/saver@eb7ae0d.

**Review by dav3r — APPROVED** [review]
👍 👍

**Pull request merged**